src/SFA.DAS.AANHub.Application/Members/Commands/PostMemberStatus/PostMemberStatusCommandHandler.cs

```diff
@@ -0,0 +1,77 @@
+﻿using MediatR;
+using SFA.DAS.AANHub.Application.Common;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Domain.Common;
+using SFA.DAS.AANHub.Domain.Entities;
+using SFA.DAS.AANHub.Domain.Interfaces;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using System.Text.Json;
+
+namespace SFA.DAS.AANHub.Application.Members.Commands.PostMemberStatus;
+
+public class PostMemberStatusCommandHandler : IRequestHandler<PostMemberStatusCommand, ValidatedResponse<SuccessCommandResult>>
+{
+    private readonly IMembersWriteRepository _membersWriteRepository;
+    private readonly IAuditWriteRepository _auditWriteRepository;
+    private readonly IAanDataContext _aanDataContext;
+    private readonly INotificationsWriteRepository _notificationsWriteRepository;
+
+    public PostMemberStatusCommandHandler(IMembersWriteRepository membersWriteRepository, IAuditWriteRepository auditWriteRepository, IAanDataContext aanDataContext, INotificationsWriteRepository notificationsWriteRepository)
+    {
+        _membersWriteRepository = membersWriteRepository;
+        _auditWriteRepository = auditWriteRepository;
+        _aanDataContext = aanDataContext;
+        _notificationsWriteRepository = notificationsWriteRepository;
+    }
+
+    public async Task<ValidatedResponse<SuccessCommandResult>> Handle(PostMemberStatusCommand request,
+        CancellationToken cancellationToken)
+    {
+        var member = await _membersWriteRepository.Get(request.MemberId);
+
+        var audit = new Audit()
+        {
+            Action = "Post",
+            Before = JsonSerializer.Serialize(member),
+            ActionedBy = request.RequestedByMemberId,
+            AuditTime = DateTime.UtcNow,
+            Resource = nameof(Member),
+            EntityId = member!.Id
+        };
+
+        member.Status = request.Status.ToString();
+        member.EndDate = DateTime.UtcNow;
+        member.LastUpdatedDate = DateTime.UtcNow;
+
+        audit.After = JsonSerializer.Serialize(member);
+        _auditWriteRepository.Create(audit);
+
+        if (request.Status == MemberRemovalStatus.Removed)
+        {
+            await CreateNotificationForRemoved(request, member);
+        }
+
+        await _aanDataContext.SaveChangesAsync(cancellationToken);
+
+        return new ValidatedResponse<SuccessCommandResult>(new SuccessCommandResult());
+
+    }
+
+    private async Task CreateNotificationForRemoved(PostMemberStatusCommand command, Member member)
+    {
+        const string templateName = Domain.Common.Constants.EmailTemplateName.AANMemberRemoved;
+        var tokens = await GetTokens(member.FullName);
+        var notification = NotificationHelper.CreateNotification(Guid.NewGuid(), command.RequestedByMemberId, templateName, tokens, command.RequestedByMemberId, true, command.MemberId.ToString());
+        _notificationsWriteRepository.Create(notification);
+    }
+
+    private static Task<string> GetTokens(string fullName)
```

Comment:
Why are we returning a Task here?

Reply:
I lifted from elsewhere and didn't think it through, I'll change to string